ratis-server/src/test/java/org/apache/ratis/RaftAsyncTests.java

```diff
@@ -402,6 +407,46 @@ void runTestAppendEntriesTimeout(CLUSTER cluster) throws Exception {
     RaftServerConfigKeys.RetryCache.setExpiryTime(getProperties(), oldExpiryTime);
   }
 
+  @Test
+  public void testCheckLeadershipFailure() throws Exception {
+    runWithNewCluster(NUM_SERVERS, this::runTestCheckLeadershipFailure);
+  }
+
+  void runTestCheckLeadershipFailure(CLUSTER cluster) throws Exception {
```

Comment:
This test can also pass when followers turn to candidate state and trigger an election. How about we delay REQUEST_VOTE in the test? And then check the state of current leader before and after timeout?

Reply:
"This test can also pass when followers turn to candidate state and trigger an election."
We don't need to worry about this, the `outstandingOp` will prevent follower to timeout.